contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java

```diff
@@ -180,23 +184,46 @@ public int makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
 		return -5;
 	}
 	
-	public String getGlobalTurnitinLTIToolId(){
-		if(globalId == null){
-			log.debug("Setting global TII LTI tool id");
-			List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
-			if ( tools == null || tools.size() < 1  || tools.size() > 1) {
-				if(tools == null)
-					log.warn("No tools found");
-				else
-					log.warn("Found: " + tools.size());				
-				log.error("getGlobalTurnitinLTIToolId: wrong global TII LTI tool configuration");
-				return null;
+	public boolean obtainGlobalTurnitinLTIToolData(){
+		log.debug("Setting global TII LTI tool properties");
+		List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
```

Comment:
Why not move it to somewhere that both the tool and the impl and access?
Make it part of the API.

On 6 May 2016 at 12:23, bgarciaentornos <notifications@github.com> wrote:

> In
> contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java
> <https://github.com/ox-it/wl-turnitin/pull/17#discussion_r62317598>:
>
> > @@ -180,23 +184,46 @@ public int makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
> >  		return -5;
> >  	}
> >  	
> > -	public String getGlobalTurnitinLTIToolId(){
> > -		if(globalId == null){
> > -			log.debug("Setting global TII LTI tool id");
> > -			List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
> > -			if ( tools == null || tools.size() < 1  || tools.size() > 1) {
> > -				if(tools == null)
> > -					log.warn("No tools found");
> > -				else
> > -					log.warn("Found: " + tools.size());				
> > -				log.error("getGlobalTurnitinLTIToolId: wrong global TII LTI tool configuration");
> > -				return null;
> > +	public boolean obtainGlobalTurnitinLTIToolData(){
> > +		log.debug("Setting global TII LTI tool properties");
> > +		List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
>
> Definitely. But we need it in both places (impl should not be included in
> the tool). 2016-05-06 13:18 GMT+02:00 Matthew Buckett <
> notifications@github.com>:
> … <#m_5664112745550056543_>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ox-it/wl-turnitin/pull/17/files/25e0348519e93241319d2ede100c98570a883e31#r62317598>
>



-- 
  Matthew Buckett, VLE Developer, IT Services, University of Oxford